deebot_client/messages/xml/clean.py

```diff
@@ -0,0 +1,161 @@
+                    event_bus.notify(StateEvent(State.CLEANING))
+                    event_reported = True
+                case CleanAction.PAUSE:
+                    event_bus.notify(StateEvent(State.PAUSED))
+                    event_reported = True
+                case CleanAction.STOP:
+                    event_bus.notify(StateEvent(State.IDLE))
+                    event_reported = True
+
+        if event_reported:
+            return HandlingResult.success()
+        return HandlingResult.analyse()
+
+
+class CleanReportServer(XmlMessage):
+    """CleanReportServer message."""
+
+    NAME = "CleanReportServer"
+
+    @classmethod
+    def _handle_xml(cls, event_bus: EventBus, xml: Element) -> HandlingResult:
+        """Handle xml message and notify the correct event subscribers.
+
+        b"<ctl ts='1744467262312' td='CleanReportServer' act='s' type='auto' cs='1134230540'/>"
+        b"<ctl ts='1744467393682' td='CleanReportServer' act='h' type='auto' sts='1744467262' cs='1134230540' area='1' last='76' mapCount='6'/>"
+
+        :return: A message response
+        """
+        event_reported = False
+        if (act := xml.attrib.get("act")) is not None:
+            if (last := xml.attrib.get("last")) is not None and last.isdecimal():
+                last_int = int(last)
+            else:
+                last_int = None
+
+            if (area := xml.attrib.get("area")) is not None and area.isdecimal():
+                area_int = int(area)
+            else:
+                area_int = None
+
+            clean_session = xml.attrib.get("cs")
+            clean_type = xml.attrib.get("type")
+
+            clean_action = CleanAction.from_xml(act)
+            if clean_action == CleanAction.STOP:
+                event_bus.notify(
+                    StatsEvent(area=area_int, time=last_int, type=clean_type)
+                )
+                event_reported = True
+            if clean_session:
+                job_status = CleanJobStatus.NO_STATUS
+                match clean_action:
+                    case CleanAction.STOP:
+                        job_status = CleanJobStatus.FINISHED
+                    case CleanAction.START | CleanAction.RESUME:
+                        job_status = CleanJobStatus.CLEANING
+                    case CleanAction.PAUSE:
+                        job_status = CleanJobStatus.MANUALLY_STOPPED
+
+                event_bus.notify(
+                    ReportStatsEvent(
+                        area=area_int,
+                        time=last_int,
+                        type=clean_type,
+                        cleaning_id=clean_session,
+                        status=job_status,
+                        content=[],
+                    )
+                )
+                event_reported = True
+
+        if event_reported:
+            return HandlingResult.success()
+        return HandlingResult.analyse()
+
+
+class CleanedPos(Pos):
```

Comment:
What is the difference between this one and the position of the bot?

Reply:
One is fired during cleaning the other if the bot is moving around (relocating)